src/nvim/version.c

```diff
@@ -2715,8 +2715,11 @@ void list_version(void)
 /// Whether it still is not too late to show an intro message
 bool may_show_intro(void)
 {
-  return (buf_is_empty(firstwin->w_buffer)
-          && (firstwin->w_buffer->b_fname == NULL)
+  return (buf_is_empty(curbuf)
+          && (curbuf->b_fname == NULL)
+          && (curbuf->handle == 1)
+          && (curwin->handle == LOWEST_WIN_ID)
+          && (!curwin->w_floating)
           && one_window(firstwin)
```

Comment:
This is equivalent to
```suggestion
          && one_window(curwin)
```